mlflow/store/tracking/file_store.py

```diff
@@ -1319,3 +1326,363 @@ def _read_helper(root, file_name, attempts_remaining=2):
+            experiment_id: String id of the experiment for this run.
+            timestamp_ms: Start time of the trace, in milliseconds since the UNIX epoch.
+            request_metadata: Metadata of the trace.
+            tags: Tags of the trace.
+
+        Returns:
+            The created TraceInfo object.
+        """
+        request_id = request_metadata.get(SpanAttributeKey.REQUEST_ID)
+        if request_id is None:
+            raise MlflowException("Request ID is required to start a trace.", INTERNAL_ERROR)
+        _validate_experiment_id(experiment_id)
+        experiment_dir = self._get_experiment_path(
+            experiment_id, view_type=ViewType.ACTIVE_ONLY, assert_exists=True
+        )
+        mkdir(experiment_dir, FileStore.TRACES_FOLDER_NAME)
+        traces_dir = os.path.join(experiment_dir, FileStore.TRACES_FOLDER_NAME)
+        mkdir(traces_dir, request_id)
+        trace_dir = os.path.join(traces_dir, request_id)
+        trace_info = TraceInfo(
+            request_id=request_id,
+            experiment_id=experiment_id,
+            timestamp_ms=timestamp_ms,
+            execution_time_ms=None,
+            status=TraceStatus.IN_PROGRESS,
+            request_metadata=request_metadata,
+            tags=tags,
+        )
+        self._save_trace_info(trace_info, trace_dir)
+        return trace_info
+
+    def _save_trace_info(self, trace_info: TraceInfo, trace_dir, overwrite=False):
+        # Save basic trace info to META_DATA_FILE_NAME
+        trace_info_dict = self._convert_trace_info_to_dict(trace_info)
+        write_yaml(trace_dir, FileStore.META_DATA_FILE_NAME, trace_info_dict, overwrite=overwrite)
+        # Save request_metadata to its own folder
+        self._write_dict_to_trace_sub_folder(
+            trace_dir,
+            FileStore.TRACE_REQUEST_METADATA_FOLDER_NAME,
+            trace_info.request_metadata,
+            overwrite=overwrite,
+        )
+        # Save tags to its own folder
+        self._write_dict_to_trace_sub_folder(
+            trace_dir, FileStore.TRACE_TAGS_FOLDER_NAME, trace_info.tags, overwrite=overwrite
+        )
+
+    def _convert_trace_info_to_dict(self, trace_info: TraceInfo):
+        """
+        Convert trace info to a dictionary for persistence.
+        Drop request_metadata and tags as they're saved into separate files.
+        """
+        trace_info_dict = asdict(trace_info)
+        trace_info_dict["status"] = trace_info.status.value
+        trace_info_dict.pop("request_metadata", None)
+        trace_info_dict.pop("tags", None)
+        return trace_info_dict
+
+    def _convert_dict_to_trace_info(self, trace_info_dict):
+        """
+        Convert trace info dictionary to TraceInfo object.
+        """
+        trace_info_dict["status"] = TraceStatus(trace_info_dict["status"])
+        return TraceInfo(**trace_info_dict)
+
+    def _write_dict_to_trace_sub_folder(self, trace_dir, sub_folder, dictionary, overwrite=False):
+        if overwrite:
```

Comment:
Do we need to throw here? 
What if we inverted this pattern? 
Current:
- make directory, save file if data exists on start_trace()
- on end_trace, enable overwrite mode, check if directory exists, overwrite if exists. 

Potential:
- `makedirs(path, exist_ok=True)`
- write data

Since the end_trace logic reads if files exist in the directory structure from the start_trace(), we can safely just write to this directory whether it exists or not with the state of the data structure when calling this function, right?

Reply:
That makes sense, if the traceInfo doesn't exist it will throw before writing!